packages/react-hooks/src/useSetInterval.test.tsx

```diff
@@ -11,13 +11,10 @@ describe('useSetInterval', () => {
     const { setInterval, clearInterval } = useSetInterval();
     const { current: state } = React.useRef<{ id: number }>({ id: 0 });
 
-    React.useImperativeHandle(
-      ref,
-      () => ({
-        clearInterval: () => clearInterval(state.id),
-      }),
-      [clearInterval],
-    );
+    // This should not have a dependency array because `clearInterval` will always be the same
```

Comment:
I can't find confirmation in docs; however, if this acts like `useEffect`, then if you don't pass a dependency array you'll end up re-assigning the ref every render. If you want to do it only once, pass an empty deps array.

Reply:
I think I misinterpreted the plugin's warning message here. Updated to remove the comment and (similar to other places) add a dependency array `[clearInterval, state]` even though neither will change, to avoid disabling the rule.